idaes/models/properties/general_helmholtz/helmholtz_state.py

```diff
@@ -0,0 +1,1115 @@
+                    doc="Vapor mole fraction (mol vapor/mol total)",
+                )
+            self._state_vars_dict = {
+                "flow_mass": self.flow_mass,
+                "energy_internal_mass": self.energy_internal_mass,
+                "pressure": self.pressure,
+            }
+            self.extensive_set = ComponentSet((self.flow_mass,))
+            self.intensive_set = ComponentSet(
+                (self.energy_internal_mass, self.pressure)
+            )
+        if self.state_vars == StateVars.TPX:
+            self.temperature = pyo.Var(
+                domain=pyo.PositiveReals,
+                initialize=params.default_temperature_value,
+                doc="Temperature",
+                bounds=params.default_temperature_bounds,
+                units=pyunits.K,
+            )
+            if phase_set == PhaseType.MIX or phase_set == PhaseType.LG:
+                self.vapor_frac = pyo.Var(
+                    initialize=0.0,
+                    units=pyunits.dimensionless,
+                    doc="Vapor fraction"
+                    # No bounds here, since it is often (usually) on it's bound
+                    # and that's not the best for IPOPT
+                )
+                self.intensive_set = ComponentSet(
+                    (self.temperature, self.pressure, self.vapor_frac)
+                )
+                self._state_vars_dict = {
+                    "temperature": self.temperature,
+                    "pressure": self.pressure,
+                    "vapor_frac": self.vapor_frac,
+                }
+            else:
+                self.intensive_set = ComponentSet((self.temperature, self.pressure))
+                self._state_vars_dict = {
+                    "temperature": self.temperature,
+                    "pressure": self.pressure,
+                }
+            if self.amount_basis == AmountBasis.MOLE:
+                self.extensive_set = ComponentSet((self.flow_mol,))
+                self._state_vars_dict["flow_mol"] = self.flow_mol
+            else:
+                self.extensive_set = ComponentSet((self.flow_mass,))
+                self._state_vars_dict["flow_mass"] = self.flow_mass
+
+    def _tpx_phase_eq(self):
+        # Saturation pressure
+        params = self.config.parameters
+        cmp = params.pure_component
+        eps_pu = params.smoothing_pressure_under
+        eps_po = params.smoothing_pressure_over
+        priv_plist = params.private_phase_list
+        plist = params.phase_list
+        rho_star = params.dens_mass_crit
+        # Convert presssures to kPa for external functions and nicer scaling in
+        # the compimentarity-type constraints.
+        P = self.pressure * params.uc["Pa to kPa"]
+        Psat = self.pressure_sat * params.uc["Pa to kPa"]
+        vf = self.vapor_frac
+        tau = self.tau
+        # Terms for determining if you are above, below, or at the Psat
+        self.P_under_sat = pyo.Expression(
```

Comment:
Minor comment, but should we consider prefacing this with an underscore to indicate that they are not physical properties and users generally shouldn't be looking at these?